### PR TITLE
Add `.ents_spangroup` to `Doc`

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

Add property `.ents_spangroup` to `Doc`.

Split off of https://github.com/explosion/spaCy/pull/11669.

### Types of change

New feature.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by svlandeg — APPROVED** [review]
Looks good to me, just want to run this by Adriane before we merge.

File: website/docs/api/doc.mdx
Comment (adrianeboyd) [inline website/docs/api/doc.mdx:771]: This isn't a `Doc` attribute, it should be added as a property similar to `Doc.ents`.

**Comment by adrianeboyd** [comment]
My main feeling is that this feels like it might be better as separate utility method rather than a `Doc` property, and the name doesn't seem especially clear to me?

**Comment by adrianeboyd** [comment]
On the other hand it's not that different as `Doc.ents`, but naming feels kind of weird / arbitrary, like `Doc.ents` isn't `Doc.ents_tuple`. I will think about it a bit...

**Comment by adrianeboyd** [comment]
Is the goal to be able to add this to `doc.spans` or to use it separately?

**Comment by rmitsch** [comment]
To use it separately. This change is motivated by `EntityLinker` expecting `SpanGroup` instances instead of `Iterable[Span]` from v4 onwards, so @svlandeg suggested adding something to obtain entities as `SpanGroup` directly (IIRC).

**Comment by adrianeboyd** [comment]
Doing this just for `Doc.ents` feels overly specific. It feels like it ought to be more like `spacy.util.as_spangroup(Union[Iterable[Span], whatever-else-you-can-convert, etc.])`.

**Comment by adrianeboyd** [comment]
I don't want you to be blocked on this. How about we add `Doc._ents_spangroup` for now and plan to think about the more general use of `SpanGroup` in the future?

**Comment by rmitsch** [comment]
We can do that, but it's not blocking me / the EL stuff. It's purely for convenience' sake, it's fine to just do
```python
from spacy.tokens import SpanGroup
sg = SpanGroup(doc, spans=doc.ents)
```
wherever I need it. Is `Docs._ents_spangroup` really ok for you or would you rather figure out a proper solution before merging?

**Comment by adrianeboyd** [comment]
Okay, then let's just mark this as a draft for now.

**Comment by honnibal** [comment]
I'd lean slightly against this I think. I think it's worth being pretty protective of the API surface on the `Doc`. The alternative code isn't bad I don't think.

**Comment by adrianeboyd** [comment]
Let's close this for now. We can always revisit it in the future.

**Pull request closed without merging**
